crates/sol-types/src/types/calls.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{Panic, Result, Revert, SolError};
+use alloc::vec::Vec;
+
+/// A collection of ABI-encoded call-like types. This currently includes
+/// [`SolCall`] and [`SolError`].
+///
+/// [`SolCall`]: crate::SolCall
+/// [`SolError`]: crate::SolError
+///
+/// ### Implementer's Guide
+///
+/// We do not recommend implementing this trait directly. Instead, we recommend
+/// using the [`sol`][crate::sol] proc macro to parse a Solidity contract
+/// definition.
+pub trait SolCalls: Sized {
+    /// The name of this type.
```

Comment:
this is the rust typename of the enum, yes?

Reply:
yes